lib/OpenQA/Git.pm

```diff
@@ -81,6 +81,20 @@ sub set_to_latest_master ($self, $args = undef) {
         return $self->_format_git_error($res, 'Unable to reset repository to origin/master') unless $res->{status};
     }
 
+    if ((!$self->config->{update_remote}) && (!$self->config->{update_branch})) {
```

Comment:
It would be good to add a comment explaining what this code does and more importantly *why*. (The "what" can also be covered by extracting this into a separate function with a good name.)

Reply:
Moving this to a new function can cause significant changes, since `$self->config->{update_remote}` is already being used in the same function `set_to_latest_master()`.
I prefer adding a comment to explain what this code does (maybe it can also cover _why_ as well)  ;-)

Reply:
The above reset is only done when `update_branch` and `do_cleanup` is set.
Should we also check for `do_cleanup` here?